.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,107 @@
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          # Ubuntu 22.04 128GB Storage AMI
+          ec2-image-id: ami-0ba430d4b7b64de57
+          ec2-instance-type: r6i.large
+          subnet-id: ${{ secrets.AWS_EC2_SUBNET_ID }}
+          security-group-id: ${{ secrets.AWS_EC2_SG_ID }}
+
+  benchmark:
+    name: Run the benchmarks on the runner
+    needs: start-runner # required to start the main job when the runner is ready
+    runs-on: ${{ needs.start-runner.outputs.label }} # run the job on the newly created runner
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Print runner info
+        run: |
+          printf "Runner lscpu:\n$(lscpu)\n"
+          printf "Runner lsmem:\n$(lsmem)\n"
+          printf "Runner nproc:\n$(nproc)\n"
+          printf "Runner uname:\n$(uname -a)\n"
+
+      - name: Install benchmark dependencies
+        run: |
+             sudo .install/install_script.sh
+             sudo apt install python3-pip -y
+             pip3 install --upgrade pip PyYAML setuptools
+             pip3 install redisbench-admin
+             pip3 install -r requirements.txt
+
+      - name: stress test
+        run: |
+          sudo apt install stress-ng -qqy
+          uptime
+          stress-ng -c 1 --timeout 60s --metrics-brief
+          uptime
+          stress-ng --stream 1 -t 60 --metrics-brief
+          uptime
+          stress-ng --ipsec-mb=1 -t 60 --metrics-brief
+          uptime
+
+      # TODO: remove "--no-check-certificate" when possible
+      - name: Download pre-generated indices
+        timeout-minutes: 20
+        run: wget --no-check-certificate -q -i tests/benchmark/data/hnsw_indices.txt -P tests/benchmark/data
+
+      - name: Benchmark
+        timeout-minutes: 120
+        run: bash -l -c "make benchmark"
+
+      - name: Collect results
+        run: |
+          ./tests/benchmark/benchmarks.sh | xargs -I {} redisbench-admin export     \
+          --redistimeseries_host      ${{ secrets.PERFORMANCE_RTS_HOST }}           \
+          --redistimeseries_port      ${{ secrets.PERFORMANCE_RTS_PORT }}           \
+          --redistimeseries_pass      ${{ secrets.PERFORMANCE_RTS_AUTH }}           \
+          --github_repo               ${{ github.event.repository.name }}           \
+          --github_org                ${{ github.repository_owner }}                \
+          --github_branch             ${{ github.head_ref || github.ref_name }}     \
+          --github_actor              ${{ github.triggering_actor }}                \
+          --results-format            google.benchmark                              \
+          --benchmark-result-file     {}_results.json
+
+  stop-runner:
```

Comment:
Given that this is used so often.. I wonder if you can pull it into a reusable workflow (see redis stack)